### PR TITLE
[Frame] Only add rounded styles when `topBar` exists

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1598

### WHAT is this pull request doing?

Only applies rounded frame styles when the `Frame` component has a `topBar` prop

### How to 🎩

For `WithoutATopBar`, make sure [story](https://5d559397bae39100201eedc1-agsvebvhin.chromatic.com/?path=/story/all-components-frame--without-a-top-bar) is the same with feature flag on and off

https://admin.web.web-81d9.sophie-schneider.us.spin.dev/store/shop1

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)

### CODE REVIEW

File: polaris-react/src/components/AppProvider/AppProvider.tsx [hunk 1]
```diff
@@ -138,8 +138,6 @@ export class AppProvider extends Component<AppProviderProps, State> {
     const {i18n, linkComponent} = this.props;
 
     this.setRootAttributes();
-    /* Temporary for dynamicTopBarAndReframe feature. Remove when feature flag is removed. */
-    this.setBodyStyles();
```
Comment (sophschneider) [inline polaris-react/src/components/AppProvider/AppProvider.tsx:142]: Moving changes from https://github.com/Shopify/polaris/pull/11846/files#diff-a7232fda7d9031ace7080a5155b4ddb352a1c6fdd5b75d0361a22b747f80be54 to `Frame` so I can conditionally apply based on the `topBar` prop.

**Review by kyledurand — APPROVED** [review]
This looks good! Side q. Do we have a plan for mobile nav? Looks a little awkward but not too bad overlaying the rounded app corners

image

**Comment by sophschneider** [comment]
@kyledurand yes! we'll need to round the corners of it [Figma link](https://www.figma.com/file/A8jMxmQVxJCvoeZ79Tx4c6/Dynamic-Top-bar-P.2?type=design&node-id=1%3A25971&mode=design&t=chDPpE083kaNZsWI-1). Just made [an issue](https://github.com/Shopify/polaris-internal/issues/1599) for it, thanks for noticing :D

**Pull request merged**